### PR TITLE
Issue #11263 - Reduce usage of glob lib/foo/*.jar in modules

### LINKED ISSUE #11263: Using `jetty.version` override from jetty-start does not use version for various environment libs.
**Jetty version(s)**
12.0.2

**Jetty Environment**
All

**Java version/vendor** `(use: java -version)`
Any

**OS type/version**
Any

**Description**
When providing a replacement for `jetty.version` when using jetty-start's `start.jar` the new jetty.version is not applied to existing environment modules too.

### PR DESCRIPTION
**Pull request by joakime**
* Using @foo.version@ technique and proper lib lists to allow arbitrary version updates from properties

Closes #11263

### CODE REVIEW

**Review by gregw — DISMISSED** [review]
nice (assuming it builds:)

File: jetty-ee10/pom.xml [hunk 1]
```diff
@@ -69,6 +69,11 @@
 
   <dependencyManagement>
     <dependencies>
+      <dependency>
```
Comment (janbartel) [inline jetty-ee10/pom.xml]: These lines are duplicates of lines 77-81.

**Review by janbartel — CHANGES_REQUESTED** [review]
Delete the duplicate lines then looks good.

File: jetty-ee10/pom.xml (hunk 1 above)
Comment (joakime) [reply jetty-ee10/pom.xml]: Duplicate lines removed.

**Review by janbartel — APPROVED** [review]

**Pull request merged**